### ISSUE TITLE
Blackout area is offset from where it should be when upgrading to 9.7.0 from 9.5.4

### ISSUE THREAD

**Issue by sandy-farnsworth** [cypress-io/cypress#22173]
### Current behavior

Blackout is not blacking out the specified area in 9.7.0. It seems to be trying, but the blacked out area is offset from where it should be.

This is my blackout area in Cypress 9.5.4:
![blackout_9 5 4](https://user-images.githubusercontent.com/107077102/172475703-80a7e7df-2f62-4b0a-89c9-f9747cec4847.png)

I am attempting to upgrade to 9.7.0, and the wrong area is being blacked out. The tests haven't changed. 
![blackout_9 7 0](https://user-images.githubusercontent.com/107077102/172476282-264065d7-f2a0-4578-aef8-3c02241fc644.png)

### Desired behavior

Blackout the specified area. 

### Test code to reproduce
         cy.findByAutomationID(`drive`)
            .should(`be.visible`)
            .matchImageSnapshot({ blackout: [`[data-automation-id="drive-item-modified-date"]`] })
### Cypress Version

9.7.0

### Other

I also tried one or two Cypress 9.6 versions and saw the same behavior there. I am unable to upgrade past 9.5.4 since it breaks my visual tests that use blackout. It seems like blackout is still functioning, it's just blacking out the wrong area on the page. These tests have been working properly with all versions of Cypress since July, 2021.
The tests work great in Cypress 9.5.4.

**Comment by AtofStryker** [comment]
Hey @sandy-farnsworth. Thank you for creating an issue! Would you be able to provide a reproduction to demonstrate the issue you are running into? I created a reproduction repository [here](https://github.com/AtofStryker/cypress-issue-22173) that you are more than welcome to use. I was not able to reproduce it. It looks like you are using something similar to Google Drive, where I tried to create something similar. Here is the screenshot I received: ![drive screenshot -- screenshots](https://user-images.githubusercontent.com/3980464/173088995-f8b94b39-0bcf-473d-912a-5779acddd39e.png)

**Comment by sandy-farnsworth** [comment]
@AtofStryker I will do that. Did you try with multiple elements?

**Comment by sandy-farnsworth** [comment]
Thank you.

**Comment by sandy-farnsworth** [comment]
I am still investigating the issue. I found that when I run my visual tests in the cypress test runner, that blackout is working properly. When I run my tests through CI, which uses the Cypress docker container for 9.7.0, the blackout area is offset. I am trying to track down what is causing that difference.

**Comment by jordanhennell** [comment]
I have noticed the same issue after upgrading. It appears to due to using jQuery's `offset` to calculate the top and left position of the blackout element, which returns values relative to the document. But then the blackout element is positioned absolutely.

The current workaround in my project is to set `__cypress-blackout` elements to `fixed`.

**Comment by AtofStryker** [comment]
@jordanhennell I wonder if some of this got introduced in `v9.6` with the [cy.origin](https://github.com/cypress-io/cypress/pull/18075/files#diff-5360c1ddf1ffa5fc7f3f3bbcbc8913d1b8527f47b4b34dfb3dc631ea41dab001R3) PR, since we refactored/changed some of the dimension calculation code. Are you able to provide a reproducible example by chance? I can see how this is a bug, but have had some issues getting it to show up.

**Comment by jordanhennell** [comment]
Indeed it does look to have been introduced when adding cross origin support. Got a small repro here, where it's attempting to blackout the English label.

```
cy.visit("https://www.wikipedia.org/")
    .get('.central-featured') // centered (margin: 0 auto) globe with languages around it
    .screenshot({
        blackout: ['[lang=en]']
    });
```

![Wikipedia English Label Blackout](https://user-images.githubusercontent.com/4422966/174229964-60dd0a9c-408d-4e92-80ba-f669ca079ec8.png)

**Comment by AtofStryker** [comment]
@jordanhennell I got the same results from your test case. Thank you for providing that! Definitely a regression/bug.

**Comment by AtofStryker** [comment]
reproduction is up to date [here](https://github.com/AtofStryker/cypress-issue-22173)

**Comment by sandy-farnsworth** [comment]
Thank you @jordanhennell !

**Comment by chrisbreiding** [comment]
Released in [`10.3.1`](https://on.cypress.io/changelog#10-3-1).

This comment thread has been locked. If you are still experiencing this issue after upgrading to
Cypress v10.3.1, please open a new issue.
